### PR TITLE
Start or stop background expiration task upon life cycle state change

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/11919

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/MapManagedService.java [hunk 1]
```diff
@@ -56,6 +56,8 @@ public void reset() {
 
     @Override
     public void shutdown(boolean terminate) {
+        mapServiceContext.getExpirationManager().stop();
+
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/map/impl/MapManagedService.java]: Can this throw an exception?

Comment (tombujok) [reply hazelcast/src/main/java/com/hazelcast/map/impl/MapManagedService.java]: Also, won't this be covered by the Lifecycle Notification on the ExpirationManager?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/MapManagedService.java]: >Can this throw an exception?
No according to contract, it should throw.

>Also, won't this be covered by the Lifecycle Notification on the ExpirationManager?
Calling it has no harm due to the idempotency, but you are right it is redundant, i will remove it.

File: hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java [hunk 2]
```diff
@@ -182,6 +190,10 @@ public synchronized void start() {
                 scheduleWithRepetition(task, taskPeriodSeconds, taskPeriodSeconds, SECONDS);
     }
 
+    /**
+     * Ends scheduling of the task that clears expired entries.
+     * Calling this method multiple times has same effect.
+     */
     public synchronized void stop() {
```
Comment (taburet) [inline hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java]: Seems like `stop` may be private now, there are no external calls to it anymore. Not sure about `start`, the only external call is from `com.hazelcast.map.impl.MapManagedService#init`, which probably may be replaced with listening for life cycle events.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java]: done all.

Comment (taburet) [reply hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java]: Great, thank you.

**Pull request merged**